streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -68,72 +69,115 @@ private static synchronized void updateNumRecordsProduces(final int delta) {
+                    if (offsetsOfPartition.getValue().get(i) != i) {
+                        System.err.println("Offset for partition " + offsetsOfPartition.getKey() + " is not " + i + " as expected but " + offsetsOfPartition.getValue().get(i));
+                        System.err.flush();
+                    }
+                }
+                System.out.println("Max offset of partition " + offsetsOfPartition.getKey() + " is " + offsetsOfPartition.getValue().get(offsetsOfPartition.getValue().size() - 1));
             }
-            Utils.sleep(rand.nextInt(10));
-        }
-        producer.close();
-        System.out.println("Producer closed: " + numRecordsProduced + " records produced");
 
-        final Properties props = new Properties();
-        props.put(ConsumerConfig.CLIENT_ID_CONFIG, "verifier");
-        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka);
-        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-        props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.toString().toLowerCase(Locale.ROOT));
 
-        try (final KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(props)) {
-            final List<TopicPartition> partitions = getAllPartitions(consumer, "data");
-            System.out.println("Partitions: " + partitions);
-            consumer.assign(partitions);
-            consumer.seekToEnd(partitions);
+            final Properties props = new Properties();
+            props.put(ConsumerConfig.CLIENT_ID_CONFIG, "verifier");
+            props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka);
+            props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+            props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+            props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.toString().toLowerCase(Locale.ROOT));
 
-            for (final TopicPartition tp : partitions) {
-                System.out.println("End-offset for " + tp + " is " + consumer.position(tp));
+            try (final KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(props)) {
+                final List<TopicPartition> partitions = getAllPartitions(consumer, "data");
+                System.out.println("Partitions: " + partitions);
+                System.out.flush();
+                consumer.assign(partitions);
+                consumer.seekToEnd(partitions);
+
+                for (final TopicPartition tp : partitions) {
+                    System.out.println("End-offset for " + tp + " is " + consumer.position(tp));
+                    System.out.flush();
+                }
             }
+            System.out.flush();
+        } finally {
+            terminated = true;
```

Comment:
We use a `try-catch` to set the flag to make sure the shutdown hook can exit quickly even in case of failure